kpi/management/commands/sync_kobocat_perms.py

```diff
@@ -0,0 +1,160 @@
+# coding: utf-8
+from django.conf import settings
+from django.contrib.postgres.aggregates import ArrayAgg
+from django.core.exceptions import ImproperlyConfigured
+from django.core.management.base import BaseCommand
+
+from kpi.constants import PERM_FROM_KC_ONLY
+from kpi.models import Asset, ObjectPermission
+from kpi.deployment_backends.kc_access.utils import (
+    assign_applicable_kc_permissions
+)
+from kpi.deployment_backends.kc_access.shadow_models import (
+    KobocatUserObjectPermission
+)
+from kpi.management.commands.sync_kobocat_xforms import _sync_permissions
+from kpi.utils.object_permission import get_perm_ids_from_code_names
+
+
+class Command(BaseCommand):
+
+    help = (
+        "Synchronize permissions of deployed forms with KoBoCAT.\n"
+        "They are synced bidirectionally unless `--mirror-kpi` option is used."
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--asset-uid',
+            action='store',
+            dest='asset_uid',
+            default=None,
+            help="Sync only a specific asset",
+        )
+        parser.add_argument(
+            '--username',
+            action='store',
+            dest='username',
+            default=None,
+            help="Sync only a specific user's assets",
+        )
+        parser.add_argument(
+            "--chunks",
+            default=1000,
+            type=int,
+            help="Update records by batch of `chunks`.",
+        )
+        parser.add_argument(
+            '--mirror-kpi',
+            action='store_true',
+            dest='mirror_kpi',
+            default=False,
+            help='Remove all KoBoCAT permission assignments and repopulate them from KPI',
+        )
+
+        super().add_arguments(parser)
+
+    def handle(self, *args, **options):
+        self._verbosity = options['verbosity']
+        self._username = options['username']
+        self._asset_uid = options['asset_uid']
+        self._chunks = options['chunks']
+
+        if not settings.KOBOCAT_URL or not settings.KOBOCAT_INTERNAL_URL:
+            raise ImproperlyConfigured(
+                'Both KOBOCAT_URL and KOBOCAT_INTERNAL_URL must be '
+                'configured before using this command'
+            )
+        self._sync_perms(**options)
+
+    def _sync_perms(self, **options):
+        self._perm_from_kc_only_pk = get_perm_ids_from_code_names(
+            PERM_FROM_KC_ONLY
+        )
+        assets = Asset.objects.only('id', 'uid', 'owner').filter(
+            _deployment_data__active=True
```

Comment:
would this mean we're only syncing permissions for projects that haven't been archived?

Reply:
Exactly, I thought we wanted to sync only active projects. 
I'll change it.